docs/proposals/zeroconf.md

```diff
@@ -0,0 +1,142 @@
+# [ZeroConf](https://en.wikipedia.org/wiki/Zero-configuration_networking) Protocol Implementation
+
+## Goal
+
+Agent implements [Zero-configuration networking](https://en.wikipedia.org/wiki/Zero-configuration_networking) (hence 'ZeroConf'), a set of technologies that help discover devices and services using DNS-based discovery. There are 2 main elements: Multicast DNS (mDNS) and DNS-based Service Discovery (DNS-SD).
+
+While ZeroConf is often used in home networks (that don't often include regular DNS), ZeroConf is broadly applicable and is useful in IoT deployments in which devices are transient, there are many devices, developers wish to dynamically manage services on these devices.
+
```

Comment:
If you want to elaborate on the use case, maybe also mention that it enables exposing these devices to kubernetes clusters and automating deployments to utilize them.

Reply:
I see you do this later in the discovery section, so might be redundant here

Reply:
Fixed: https://github.com/deislabs/akri/pull/162/commits/7cd098506bf259749be5e57f8afd90eb2b458a84